{"cells":[{"metadata":{"trusted":true},"cell_type":"code","source":"\"\"\"Q1. Describe three applications for exception processing.\na = Exception processing is vey important to find exception that causes to run time error\nas runtime error haults the programme when exception occured\n1) checking appropriate use of input in application\n2)checking arithmatic exception in mathematic application\n3)cheking file i/o in file handling\"\"\"","execution_count":1,"outputs":[{"output_type":"execute_result","execution_count":1,"data":{"text/plain":"'Q1. Describe three applications for exception processing.\\na = Exception processing is vey important to find exception that causes to run time error\\nas runtime error haults the programme when exception occured\\n1) checking appropriate use of input in application\\n2)checking arithmatic exception in mathematic application\\n3)cheking file i/o in file handling'"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"\"\"\"Q2. What happens if you don't do something extra to treat an exception?\nA =if exception is not handled then it will restrict the flow of program will be broken during run time which might lead to abnormal termination of program.\ninshort inability to handel exception will crash the program\"\"\"","execution_count":2,"outputs":[{"output_type":"execute_result","execution_count":2,"data":{"text/plain":"\"Q2. What happens if you don't do something extra to treat an exception?\\nA =if exception is not handled then it will restrict the flow of program will be broken during run time which might lead to abnormal termination of program.\\ninshort inability to handel exception will crash the program\""},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"\"\"\"Q3. What are your options for recovering from an exception in your script?\nA =python provide try and except to recover from exception script\"\"\"","execution_count":3,"outputs":[{"output_type":"execute_result","execution_count":3,"data":{"text/plain":"'Q3. What are your options for recovering from an exception in your script?\\nA =python provide try and except to recover from exception script'"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"\"\"\"Q4. Describe two methods for triggering exceptions in your script.\nA =raise and assert are two method to handel the exception in python\nraise method will trigger if condition given in code comes true\nassert method triggers if condition is not true exception will be raised\"\"\"\nx = 10\nraise exception(f'raise exception as provided value of x is not 5 but,{x}')","execution_count":5,"outputs":[{"output_type":"error","ename":"NameError","evalue":"name 'exception' is not defined","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-5-6865cfaf5397>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m assert method triggers if condition is not true exception will be raised\"\"\"\n\u001b[1;32m      5\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m10\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0;32mraise\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf'raise exception as provided value of x is not 5 but,{x}'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;31mNameError\u001b[0m: name 'exception' is not defined"]}]},{"metadata":{"trusted":true},"cell_type":"code","source":"\"\"\"Q5. Identify two methods for specifying actions to be executed at termination time, regardless of\nwhether or not an exception exists.\nA =python provides else and finally block at the end of program to be executed regardless of wheather exception exist o not\"\"\"","execution_count":6,"outputs":[{"output_type":"execute_result","execution_count":6,"data":{"text/plain":"'Q5. Identify two methods for specifying actions to be executed at termination time, regardless of\\nwhether or not an exception exists.\\nA =python provides else and finally block at the end of program to be executed regardless of wheather exception exist o not'"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"name":"python3","display_name":"Python 3","language":"python"},"language_info":{"name":"python","version":"3.6.15","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat":4,"nbformat_minor":5}